data_steward/test/unit_test/data_steward/curation_logging/gcp_stackdriver_logger_test.py

```diff
@@ -0,0 +1,275 @@
+            'severity': SEVERITY_ERROR,
+            'time': self.log_record_created.isoformat(),
+            'sourceLocation': {
+                'file': self.file_path,
+                'functionName': self.file_name,
+                'line': 12
+            }
+        }
+
+        self.mock_logging_service_client_patcher = patch(
+            'curation_logging.curation_gae_handler.gcp_logging_v2.LoggingServiceV2Client')
+        self.mock_logging_service_client = self.mock_logging_service_client_patcher.start()
+
+    def tearDown(self):
+        self.mock_logging_service_client_patcher.stop()
+        self.request.stop()
+
+    @staticmethod
+    def create_log_record(record_created, levelno, funcName, pathname, lineno, msg):
+        log_record_magic = MagicMock()
+        type(log_record_magic).created = PropertyMock(return_value=record_created)
+        type(log_record_magic).levelno = PropertyMock(return_value=levelno)
+        type(log_record_magic).funcName = PropertyMock(return_value=funcName)
+        type(log_record_magic).pathname = PropertyMock(return_value=pathname)
+        type(log_record_magic).lineno = PropertyMock(return_value=lineno)
+        type(log_record_magic).msg = PropertyMock(return_value=msg)
+        return log_record_magic
+
+    @mock.patch('data_steward.curation_logging.curation_gae_handler.datetime')
+    def test_gcp_stackdriver_logger(self, mock_datetime):
+        mock_datetime.now.return_value.isoformat.return_value = self.request_start_time.isoformat()
+        mock_datetime.utcnow.return_value = self.request_start_time
+        mock_datetime.utcfromtimestamp.return_value = self.log_record_created
+
+        # Initialize GCPStackDriverLogger
+        self.gcp_stackdriver_logger = GCPStackDriverLogger(LOG_BUFFER_SIZE)
+        self.gcp_stackdriver_logger.setup_from_request(self.request)
+
+        self.assertIsNone(self.gcp_stackdriver_logger._first_log_ts)
+        self.assertEqual(self.gcp_stackdriver_logger._start_time, self.request_start_time.isoformat())
+        self.assertEqual(self.gcp_stackdriver_logger._request_method, self.request_method)
+        self.assertEqual(self.gcp_stackdriver_logger._request_resource, self.request_full_path)
+        self.assertEqual(self.gcp_stackdriver_logger._request_agent, self.request_user_agent)
+        self.assertEqual(self.gcp_stackdriver_logger._request_remote_addr, self.request_ip)
+        self.assertEqual(self.gcp_stackdriver_logger._request_host, self.request_host_name)
+        self.assertEqual(self.gcp_stackdriver_logger._request_log_id, self.request_log_id)
+        self.assertEqual(self.gcp_stackdriver_logger._trace, self.request_trace)
+
+        self.gcp_stackdriver_logger.log_event(self.info_log_record)
+        self.gcp_stackdriver_logger.log_event(self.debug_log_record)
+        self.gcp_stackdriver_logger.log_event(self.error_log_record)
+
+        self.assertEqual(len(self.gcp_stackdriver_logger._buffer), 0)
```

Comment:
Sorry, this was not explicitly mentioned in the test. That's because I set up the buffer_size to be 3, therefore it only writes once after 3 records are added

Reply:
Please add a comment or put a log into the assert to explain why you expect 0 here, e.g.

`self.assertEqual(len(self.gcp_stackdriver_logger._buffer), 0, 'expected log buffer to flush itself after being filled')`

Reply:
This has been done